pkg/sql/truncate.go

```diff
@@ -177,6 +177,19 @@ func (p *planner) truncateTable(ctx context.Context, id sqlbase.ID, traceKV bool
 	if err := newTableDesc.SetUpVersion(); err != nil {
 		return err
 	}
+	// Resolve all outstanding mutations. Make all new schema elements
+	// public because the table is empty and doesn't need to be backfilled.
+	for _, m := range newTableDesc.Mutations {
+		if m.Direction == sqlbase.DescriptorMutation_ADD {
```

Comment:
What about mutations that drop a column? These columns must be removed from the new descriptor.

Reply:
Those columns/indexes have already been removed from the descriptor and placed in the mutations. They are ignored here implying that they get dropped.

Reply:
added a test for this case